idseq_dag/steps/generate_loc_db.py

```diff
@@ -85,32 +84,48 @@ def generate_loc_db_for_sqlite(self, db_file, loc_db_file, info_db_file):
             PipelineStepGenerateLocDB.generate_loc_db_work(db_file, loc_db, info_db)
 
     @run_in_subprocess
-    def generate_loc_db_for_shelf(self, db_file, loc_db_file):
+    def generate_loc_db_for_shelf(self, db_file, loc_db_file, info_db_file):
         loc_dict = shelve.Shelf(dbm.ndbm.open(loc_db_file.replace(".db", ""), 'c'))
+        info_dict = shelve.Shelf(dbm.ndbm.open(info_db_file.replace(".db", ""), 'c'))
         with open(db_file) as dbf:
             seq_offset = 0
             seq_len = 0
+            seq_bp_len = 0
             header_len = 0
             lines = 0
             accession_id = ""
+            accession_name = ""
             for line in dbf:
                 lines += 1
                 if lines % 100000 == 0:
                     log.write(f"{lines/1000000.0}M lines")
                 if line[0] == '>':  # header line
                     if seq_len > 0 and len(accession_id) > 0:
-                        loc_dict[accession_id] = (seq_offset, header_len, seq_len)
+                        loc_dict[accession_id] = [seq_offset, header_len, seq_len]
+                    if seq_bp_len > 0 and len(accession_name) > 0:
+                        info_dict[accession_id] = [accession_name, seq_bp_len]
+
                     seq_offset = seq_offset + header_len + seq_len
                     header_len = len(line)
                     seq_len = 0
-                    s = re.match('^>([^ ]*).*', line)
+                    seq_bp_len = 0
+                    accession_name = ""
+                    # Sometimes multiple accessions will be mapped to a single sequence.
+                    # In this case, they will be separated by the \x01 char.
+                    # To get the accession name, just match until the first \x01.
```

Comment:
so are we preferring the first accession then? by what justification?

Reply:
also it seems like this is copy pasted without comment from a method above

Reply:
Your second comment is correct, I will add a comment, though it is not 100% copy pasted. The logic should be identical to the sqlite logic which is why I made this transfer. I am not sure why we made this choice, this is only concerning the format of the resulting key value store. I will add a comment.